modules/oe_content_event/src/Plugin/InternalLinkSourceFilter/EventPeriodFilter.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_content_event\Plugin\InternalLinkSourceFilter;
+
+use Drupal\Core\Cache\RefinableCacheableDependencyInterface;
+use Drupal\Core\Datetime\DrupalDateTime;
+use Drupal\Core\Entity\Query\QueryInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\datetime\Plugin\Field\FieldType\DateTimeItemInterface;
+use Drupal\oe_link_lists_internal_source\InternalLinkSourceFilterInterface;
+use Drupal\oe_link_lists_internal_source\InternalLinkSourceFilterPluginBase;
+
+/**
+ * Event link source filter class.
+ *
+ * @InternalLinkSourceFilter(
+ *   id = "oe_content_event_period",
+ *   label = @Translation("Event period link source filter"),
+ *   description = @Translation("Filters events by the period they are in: past or upcoming."),
+ *   entity_types = {
+ *     "node" = {
+ *       "oe_event",
+ *     },
+ *   },
+ * )
+ */
+class EventPeriodFilter extends InternalLinkSourceFilterPluginBase implements InternalLinkSourceFilterInterface {
+
+  /**
+   * Option for upcoming events.
+   */
+  const UPCOMING = 1;
+
+  /**
+   * Option for past events.
+   */
+  const PAST = -1;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function defaultConfiguration() {
+    return [
+      'period' => self::UPCOMING,
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function isApplicable(string $entity_type, string $bundle): bool {
```

Comment:
Is this needed? i don't think so. it's done based on the annotation. Please see InternalLinkSourceFilterPluginManager which loads the applicable filters.